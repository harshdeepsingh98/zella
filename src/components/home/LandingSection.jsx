import React from 'react';
import styled from 'styled-components';
import { Box, Typography } from '@mui/material';
import { useNavigate } from 'react-router-dom';
import { Button } from '@components/common';
import useAppSelector from '@hooks/useAppSelector';
import { selectAppMetadata } from '@features/app/selectors';
import sampleImage from '@assets/images/sample.png';

const Container = styled(Box)`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  height: 100vh;
  width: 100%;
  background-color: #000000;
  color: #ffffff;
  position: relative;
  background-image: url(${sampleImage});
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
`;

const Overlay = styled(Box)`
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: linear-gradient(0deg, rgba(0, 0, 0, 0.8) 0%, rgba(0, 0, 0, 0.4) 100%);
  z-index: 1;
`;

const Content = styled(Box)`
  display: flex;
  flex-direction: column;
  height: 100%;
  width: 100%;
  z-index: 2;
  padding: 24px;
`;

const HeaderSection = styled(Box)`
  display: flex;
  align-items: center;
  margin-top: 12px;
`;

const Logo = styled.div`
  display: flex;
  align-items: center;
  color: #33a54a;
  font-weight: 600;
  font-size: 1.5rem;

  svg {
    width: 24px;
    height: 24px;
    margin-right: 8px;
  }
`;

const LogoIcon = () => (
  <svg viewBox="0 0 24 24" fill="currentColor" xmlns="http://www.w3.org/2000/svg">
    <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm-5-9h10v2H7z" />
  </svg>
);

const MainContent = styled(Box)`
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  flex: 1;
  padding-bottom: 40px;
`;

const Title = styled(Typography)`
  font-size: 2.5rem;
  font-weight: 700;
  margin-bottom: 16px;
`;

const Subtitle = styled(Typography)`
  font-size: 1rem;
  margin-bottom: 32px;
  color: rgba(255, 255, 255, 0.9);
`;

const ButtonContainer = styled(Box)`
  width: 100%;
  margin-bottom: 16px;
`;

const AvatarCircle = styled(Box)`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 32px;
  height: 32px;
  border-radius: 50%;
  background-color: #a646ff;
  color: white;
  font-weight: 600;
  font-size: 14px;
  margin-right: 4px;
`;

const SecondAvatarCircle = styled(AvatarCircle)`
  background-color: #46a6ff;
  margin-left: -8px;
`;

const AvatarContainer = styled(Box)`
  display: flex;
  align-items: center;
  position: absolute;
  top: 16px;
  right: 16px;
`;

const LandingSection = () => {
  const navigate = useNavigate();
  const appMetadata = useAppSelector(selectAppMetadata);

  const handleGetStarted = () => {
    navigate('/auth/mobile');
  };

  return (
    <Container>
      <Overlay />

      <Content>
        <HeaderSection>
          <Logo>
            <LogoIcon />
            {appMetadata.name || 'Zella'}
          </Logo>
        </HeaderSection>

        <MainContent>
          <Title variant="h1">{appMetadata.name || 'Zella'}</Title>

          <Subtitle variant="subtitle1">
            {appMetadata.description || 'Secure digital identity platform'}
          </Subtitle>

          <ButtonContainer>
            <Button fullWidth variant="contained" size="large" onClick={handleGetStarted}>
              Get Started
            </Button>
          </ButtonContainer>
        </MainContent>
      </Content>
    </Container>
  );
};

export default LandingSection;
