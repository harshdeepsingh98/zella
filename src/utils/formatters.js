// Format mobile number
export const formatMobileNumber = (mobileNumber) => {
    if (!mobileNumber) return '';
    
    // Check if it's a string
    if (typeof mobileNumber !== 'string') {
      mobileNumber = String(mobileNumber);
    }
    
    // Remove any spaces or special characters
    mobileNumber = mobileNumber.replace(/\s+/g, '');
    
    // Format: +91 98765 43210
    if (mobileNumber.length === 10) {
      return `+91 ${mobileNumber.substring(0, 5)} ${mobileNumber.substring(5)}`;
    }
    
    return `+91 ${mobileNumber}`;
  };
  
  // Format time from seconds to MM:SS
  export const formatTime = (seconds) => {
    if (!seconds || seconds < 0) return '00:00';
    
    const minutes = Math.floor(seconds / 60);
    const remainingSeconds = seconds % 60;
    
    return `${minutes.toString().padStart(2, '0')}:${remainingSeconds.toString().padStart(2, '0')}`;
  };
  
  // Format ISO duration string (PT60S) to seconds
  export const formatDuration = (isoDuration) => {
    if (!isoDuration) return 0;
    
    // Handle PT60S format (60 seconds)
    const secondsMatch = isoDuration.match(/PT(\d+)S/);
    if (secondsMatch && secondsMatch[1]) {
      return parseInt(secondsMatch[1], 10);
    }
    
    // Handle PT1M format (1 minute)
    const minutesMatch = isoDuration.match(/PT(\d+)M/);
    if (minutesMatch && minutesMatch[1]) {
      return parseInt(minutesMatch[1], 10) * 60;
    }
    
    // Handle PT1H format (1 hour)
    const hoursMatch = isoDuration.match(/PT(\d+)H/);
    if (hoursMatch && hoursMatch[1]) {
      return parseInt(hoursMatch[1], 10) * 60 * 60;
    }
    
    return 0;
  };
  
  // Format template strings with variables
  export const formatTemplate = (template, variables) => {
    if (!template) return '';
    if (!variables || typeof variables !== 'object') return template;
    
    let result = template;
    
    // Replace all occurrences of {{ variable_name }}
    Object.keys(variables).forEach(key => {
      const regex = new RegExp(`{{\\s*${key}\\s*}}`, 'g');
      result = result.replace(regex, variables[key]);
    });
    
    return result;
  };