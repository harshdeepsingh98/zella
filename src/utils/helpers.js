import { STORAGE_KEYS } from '@config/constants';

// Local storage helpers
export const storage = {
  get: (key) => {
    try {
      const value = localStorage.getItem(key);
      return value ? JSON.parse(value) : null;
    } catch (error) {
      console.error('Error retrieving from localStorage:', error);
      return null;
    }
  },
  
  set: (key, value) => {
    try {
      localStorage.setItem(key, JSON.stringify(value));
      return true;
    } catch (error) {
      console.error('Error saving to localStorage:', error);
      return false;
    }
  },
  
  remove: (key) => {
    try {
      localStorage.removeItem(key);
      return true;
    } catch (error) {
      console.error('Error removing from localStorage:', error);
      return false;
    }
  },
  
  clear: () => {
    try {
      localStorage.clear();
      return true;
    } catch (error) {
      console.error('Error clearing localStorage:', error);
      return false;
    }
  },
};

// Authentication helpers
export const auth = {
  getToken: () => storage.get(STORAGE_KEYS.AUTH_TOKEN),
  
  setToken: (token) => storage.set(STORAGE_KEYS.AUTH_TOKEN, token),
  
  removeToken: () => storage.remove(STORAGE_KEYS.AUTH_TOKEN),
  
  isAuthenticated: () => !!storage.get(STORAGE_KEYS.AUTH_TOKEN),
  
  getUserId: () => storage.get(STORAGE_KEYS.USER_ID),
  
  setUserId: (userId) => storage.set(STORAGE_KEYS.USER_ID, userId),
  
  logout: () => {
    storage.remove(STORAGE_KEYS.AUTH_TOKEN);
    storage.remove(STORAGE_KEYS.USER_ID);
  },
};

// Language helpers
export const language = {
  getLanguage: () => storage.get(STORAGE_KEYS.LANGUAGE) || 'en',
  
  setLanguage: (lang) => storage.set(STORAGE_KEYS.LANGUAGE, lang),
};

// Debounce function
export const debounce = (func, wait) => {
  let timeout;
  
  return function executedFunction(...args) {
    const later = () => {
      clearTimeout(timeout);
      func(...args);
    };
    
    clearTimeout(timeout);
    timeout = setTimeout(later, wait);
  };
};

// Generate random ID
export const generateId = (prefix = 'id') => {
  return `${prefix}-${Math.random().toString(36).substring(2, 9)}`;
};

// Deep clone object
export const deepClone = (obj) => {
  if (obj === null || typeof obj !== 'object') return obj;
  
  try {
    return JSON.parse(JSON.stringify(obj));
  } catch (error) {
    console.error('Error cloning object:', error);
    return obj;
  }
};