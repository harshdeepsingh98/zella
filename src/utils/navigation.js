// Place this in a utility file like src/utils/navigation.js

/**
 * Find the next page in the flow based on the current page code
 * @param {Array} pages - Array of page configurations from Redux
 * @param {String} currentPageCode - The code of the current page
 * @returns {Object|null} - The next page object or null if not found
 */
export const findNextPage = (pages, currentPageCode) => {
  if (!pages || pages.length === 0) return null;

  // Find the index of the current page
  const currentIndex = pages.findIndex(page => page.metadata.code === currentPageCode);

  // If found and not the last page, return the next page
  if (currentIndex !== -1 && currentIndex < pages.length - 1) {
    return pages[currentIndex + 1];
  }

  return null;
};

/**
 * Find the previous page in the flow based on the current page code
 * @param {Array} pages - Array of page configurations from Redux
 * @param {String} currentPageCode - The code of the current page
 * @returns {Object|null} - The previous page object or null if not found
 */
export const findPrevPage = (pages, currentPageCode) => {
  if (!pages || pages.length === 0) return null;

  // Find the index of the current page
  const currentIndex = pages.findIndex(page => page.metadata.code === currentPageCode);

  // If found and not the first page, return the previous page
  if (currentIndex > 0) {
    return pages[currentIndex - 1];
  }

  return null;
};
