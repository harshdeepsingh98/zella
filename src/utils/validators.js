import { VALIDATION } from '@config/constants';

// Validate mobile number
export const validateMobileNumber = (mobileNumber) => {
  if (!mobileNumber) return false;
  
  // Check if it's a string
  if (typeof mobileNumber !== 'string') {
    mobileNumber = String(mobileNumber);
  }
  
  // Remove any spaces or special characters
  mobileNumber = mobileNumber.replace(/\s+/g, '');
  
  // Check if it matches the pattern
  return VALIDATION.MOBILE.PATTERN.test(mobileNumber);
};

// Validate OTP
export const validateOtp = (otp) => {
  if (!otp) return false;
  
  // Check if it's a string
  if (typeof otp !== 'string') {
    otp = String(otp);
  }
  
  // Remove any spaces
  otp = otp.replace(/\s+/g, '');
  
  // Check if it matches the pattern
  return VALIDATION.OTP.PATTERN.test(otp);
};

// Validate name
export const validateName = (name) => {
  if (!name) return false;
  
  // Check if it's a string
  if (typeof name !== 'string') return false;
  
  // Trim the name
  name = name.trim();
  
  // Check length
  if (name.length < VALIDATION.NAME.MIN_LENGTH || name.length > VALIDATION.NAME.MAX_LENGTH) {
    return false;
  }
  
  return true;
};

// Validate email
export const validateEmail = (email) => {
  if (!email) return false;
  
  // Check if it's a string
  if (typeof email !== 'string') return false;
  
  // Simple email regex
  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
  
  return emailRegex.test(email);
};

// Generic field length validator
export const validateFieldLength = (value, minLength, maxLength) => {
  if (!value) return minLength === 0;
  
  // Check if it's a string
  if (typeof value !== 'string') {
    value = String(value);
  }
  
  // Check length
  const length = value.trim().length;
  
  if (minLength !== undefined && length < minLength) {
    return false;
  }
  
  if (maxLength !== undefined && length > maxLength) {
    return false;
  }
  
  return true;
};

// Validate by type
export const validateByType = (value, type, options = {}) => {
  switch (type) {
    case 'mobileNumber':
      return validateMobileNumber(value);
      
    case 'otp':
      return validateOtp(value);
      
    case 'name':
      return validateName(value);
      
    case 'email':
      return validateEmail(value);
      
    case 'text':
      return validateFieldLength(value, options.minLength, options.maxLength);
      
    default:
      return true;
  }
};