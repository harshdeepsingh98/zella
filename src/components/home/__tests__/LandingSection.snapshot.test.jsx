// src/components/home/__tests__/LandingSection.snapshot.test.jsx
import { describe, it, expect, vi } from 'vitest';
import { renderWithMui } from '../../../test/utils/mui-test-utils';
import LandingSection from '../LandingSection';

// Mock the useNavigate hook
vi.mock('react-router-dom', async () => {
  const actual = await vi.importActual('react-router-dom');
  return {
    ...actual,
    useNavigate: () => vi.fn(),
  };
});

// Only mock useAppSelector to return what we need
vi.mock('@hooks/useAppSelector', () => ({
  default: vi.fn(selector => {
    // This simulates the selector function by returning different values
    // based on which selector is being called
    if (selector.name === 'selectAppMetadata') {
      return { name: 'Snapshot App', description: 'Snapshot Description' };
    }
    if (selector.name === 'selectPages') {
      return [{ metadata: { code: 'test-page' } }];
    }
    return null;
  }),
}));

describe('LandingSection Snapshots', () => {
  it('renders with default props', () => {
    // Create initial state
    const initialState = {
      app: {
        data: {
          app: {
            metadata: {
              name: 'Snapshot App',
              description: 'Snapshot Description',
            },
            config: {
              pages: [
                {
                  metadata: {
                    name: 'Test Page',
                    code: 'test-page',
                  },
                },
              ],
            },
          },
        },
      },
    };

    // Use the MUI-compatible render function
    const { container } = renderWithMui(<LandingSection />, { initialState });

    // Test that it renders without crashing
    expect(container).toBeTruthy();
  });
});
