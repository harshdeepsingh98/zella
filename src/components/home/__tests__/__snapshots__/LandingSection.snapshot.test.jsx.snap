// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LandingSection Snapshots > matches snapshot with default props 1`] = `
<DocumentFragment>
  <div
    class="sc-fjUQFl fsNohD MuiBox-root css-0"
  >
    <div
      class="sc-guGTOK cxBxXY MuiBox-root css-0"
    />
    <div
      class="sc-hZARmv jfFzUi MuiBox-root css-0"
    >
      <div
        class="sc-dKKIkQ kddPpa MuiBox-root css-0"
      >
        <div
          class="sc-fpEFIB bZpxwD"
        >
          <svg
            fill="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm-5-9h10v2H7z"
            />
          </svg>
          Snapshot App
        </div>
      </div>
      <div
        class="sc-gAqISa hziRZQ MuiBox-root css-0"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 sc-ifkGpL kCMiLX css-o2w69a-MuiTypography-root"
        >
          Snapshot App
        </h1>
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 sc-fbguzk gbCBWz css-10wpov9-MuiTypography-root"
        >
          Snapshot Description
        </h6>
        <div
          class="sc-gGarWV jVTCia MuiBox-root css-0"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary MuiButton-fullWidth sc-Qotzb eHpXPy css-189e9pj-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Get Started
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
